### ISSUE TITLE
`@conda` decorator incompatible with newer versions of conda

### ISSUE THREAD

**Issue by turambar** [Netflix/metaflow#1496]
Hello Metafriends! It appears that the `@conda` decorator is incompatible with newer versions of conda.

When attempting to run a simple flow with the decorator using the `--environment=conda` flag, I received a JSONDecodeError. Jan P in the #ask-metaflow channel pointed out that in newer versions of conda, the `conda info` command (and `self._info()` method) no longer returns a JSON formatted response, which Metaflow's conda internals expect. The interim fix was to downgrade conda to `22.11.1`.

Here are some rough details to hopefully enable y'all to reproduce:

- Install the latest version of Miniconda (I used the [MacOSX arm version](https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-arm64.sh))
- Create a simple flow, e.g., `test_flow.py` using the `@conda` decorator
- Run flow with `python test_flow.py --environment=conda run`
- Got the decoding error (see below)
- Uninstall miniconda and re-install using an older installer for 22.11.1 from the archive ([link to Mac version](https://repo.anaconda.com/miniconda/Miniconda3-py310_22.11.1-1-MacOSX-arm64.sh)).
- Try re-running flow as above
- Success!

Here's the original error:

```
(mflow2) davidkale@ringil metaflow-tools % python test_flow.py --environment=conda run
Metaflow 2.9.11 executing TestFlow for user:davidkale
Validating your flow...
    The graph looks good!
Running pylint...
    Pylint not found, so extra checks are disabled.
Bootstrapping conda environment...(this could take a few minutes)
    Internal error
Traceback (most recent call last):
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/cli.py", line 1172, in main
    start(auto_envvar_prefix="METAFLOW", obj=state)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/_vendor/click/core.py", line 829, in __call__
    return self.main(args, kwargs)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/_vendor/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/_vendor/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/_vendor/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, ctx.params)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/_vendor/click/core.py", line 610, in invoke
    return callback(args, kwargs)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/cli.py", line 691, in wrapper
    return func(args, kwargs)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/_vendor/click/decorators.py", line 33, in new_func
    return f(get_current_context().obj, args, kwargs)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/cli.py", line 828, in run
    before_run(obj, tags, decospecs + obj.environment.decospecs())
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/cli.py", line 886, in before_run
    obj.package = MetaflowPackage(
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/package.py", line 73, in __init__
    deco.package_init(flow, step.__name__, environment)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/plugins/conda/conda_step_decorator.py", line 355, in package_init
    self._prepare_step_environment(step, self.local_root)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/plugins/conda/conda_step_decorator.py", line 216, in _prepare_step_environment
    env_id = self._resolve_step_environment(ds_root)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/plugins/conda/conda_step_decorator.py", line 124, in _resolve_step_environment
    CondaStepDecorator.conda = Conda()
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/plugins/conda/conda.py", line 43, in __init__
    if LooseVersion(self._info()["conda_version"]) < LooseVersion("4.6.0"):
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/site-packages/metaflow/plugins/conda/conda.py", line 111, in _info
    return json.loads(self._call_conda(["info"]))
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/Users/davidkale/miniconda3/envs/mflow2/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by nanohanno** [comment]
I observed the same behaviour. It seems to be raised by a regression in conda which has been fixed but not yet released https://github.com/conda/conda/issues/12987

**Comment by csetzkorn** [comment]
me too ..

**Comment by savingoyal** [comment]
#1505 will fix this.

**Comment by savingoyal** [comment]
fixed in #1505
